sdk/core/azure-core/src/main/java/com/azure/core/models/ResponseError.java

```diff
@@ -117,4 +110,134 @@ ResponseError setErrorDetails(List<ResponseError> errorDetails) {
         this.errorDetails = errorDetails;
         return this;
     }
+
+    @Override
+    public StringBuilder toJson(StringBuilder builder) {
```

Comment:
Since `ResponseError` is a public type that users inspect, do we want to have serialization APIs (toJson and fromJson) to be visible to the users? Also, each time a new property is added these methods have to be updated to handle the new properties. Is there a way to centralize this and not have to do this for all JsonCapable types?

Reply:
For now, I want to keep this APIs public to allow customers who may have been using our models in serialization new options when we remove databind support.